cmd/globals.go

```diff
@@ -177,6 +177,12 @@ var (
 	colorRed    = color.New(color.FgRed).SprintfFunc()
 )
 
+var (
+	globalQuiet     = false
+	globalJSON      = false
```

Comment:
Why do you need this logger specific flag as global?

Reply:
because you need to support flags for all commands i.e `--json` should be honored

```
minio --json gateway s3
minio gateway --json s3
minio gateway s3 --json
```

The approach taken is similar to how `mc` does for all the certain globally applicable flags.

Reply:
Who is using this flag other than log?

Reply:
This value is used by each commands to turn a specific feature on it can be JSON or Quiet. Now the position of the `flag` should honor it in a way globally because it is a globalFlag afterall. We call these flags globalFlags for this reason alone they should be position independent and should be looked into the context correctly. 

Current code has a bug and doesn't work since the interpretation is wrong. Command line flags are only interpreted at sub-command level.

Reply:
Unable to understand the problem.

As these flags are used in logger (only), why do we need as a global?  In what are other cases we need this flag?  What is the bug you are referring to these flag?

Another question is why do we want to change to fix issue https://github.com/minio/minio/issues/5403?

Reply:
I guess you should talk to @kannappanr he will fill you in.

Reply:
@balamurugana : Currently, we would expect that `minio --arg=value gateway s3` to behave same as `minio gateway s3 --arg=value`. But, in the first case, the context is not completely passed to the subcommand, which in this case is gateway-s3. Only the argument name is passed. The value is not passed. Even though, `--json` and `quiet` are flags and are not arguments, we wanted this to be generalized and made it part of this function.